doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,145 @@
+    - `record` types where all fields types are supported.
+
+We intend to relax some of these restrictions in the future.
+
+## Examples
+
+These examples create real objects. After you have tried the examples, make sure to drop these objects and spin down any resources you may have created.
+
+### Partitioning by timestamp
+
+For timeseries or "event"-type collections, it's often useful to partition the data by timestamp.
+
+1. First, create a table called `events`.
+    ```mzsql
+    -- Create a table of timestamped events. Note that the `event_ts` column is
+    -- first in the column list and in the parition-by clause.
+    CREATE TABLE events (
+        event_ts timestamptz,
+        content text
+    ) WITH (
+        PARTITION BY (event_ts)
+    );
+    ```
+
+1. Insert a few records, one "older" record and one more recent.
+    ```mzsql
+    INSERT INTO events VALUES (now()::timestamp - '1 minute', 'hello');
+    INSERT INTO events VALUES (now(), 'world');
+    ```
+
+1. If we run a select statement against the data sometime in the next couple of minutes, we return one row but not the other.
+    ```mzsql
+    SELECT * FROM events WHERE event_ts > mz_now() - '2 minutes';
+    ```
+
+1. An `EXPLAIN FILTER PUSHDOWN` statement shows that we're able to fetch a subset of the parts in our collection... only those parts that contain data with recent timestamps.
+    ```mzsql
+    EXPLAIN FILTER PUSHDOWN FOR
+    SELECT * FROM events WHERE event_ts + '2 minutes' > mz_now();
+    ```
+
+If you wait a few minutes longer until there are no events that match the temporal filter, you'll notice that not only does the query return zero rows, but the explain shows that we fetched zero parts.
+
+Note that the exact numbers you see here may very: parts can be much larger than a single row, and the actual level of filtering may vary for small datasets as data is compacted together internally. However, datasets of a few gigabytes or larger should reliably see benefits from this optimization.
+
+### Partitioning by primary key
+
+Other datasets don't have a strong timeseries component, but they do have a clear notion of type or category. For example, let's suppose we manage a collection of music venues spread across the world, but regularly want to target queries to just those that exist in a single country.
+
+1. First, create a table called `venues`, partitioned by country.
+    ```mzsql
+    -- Create a table for our venue data.
+    -- Once again, the partition column is listed first.
+    CREATE TABLE venues (
+        country_code text,
+        id uuid,
+    ) WITH (
+        PARTITION BY (country_code)
+    );
+    ```
+
+1. Insert a few records, one "older" record and one more recent.
+    ```mzsql
+    INSERT INTO venues VALUES ('US', uuid_generate_v5('venue', 'Rock World'));
+    INSERT INTO venues VALUES ('CA', uuid_generate_v5('venue', 'Friendship Cove'));
+    ```
+
+1. We can filter down our list of venues to just those in a specific country.
```

Comment:
```suggestion
1. Query by a specific country.
```
Also, should we also show an example where if we had inserted data for  'US', 'CA', 'MX' and query for ('CA', 'MX')

Reply:
I would expect that only the parts with CA and MX will return (they could possibly be the same parts if I guess we don't have a lot of data between CA and MX -- but not really something that concerns me here) ... but it's more that for some of our users who may not have deep SQL background, I don't want them to think they have to issue SELECT statement per country - but that partition can also help if you do `SELECT * FROM venues WHERE country_code in ('CA', 'MX')`.  That is, I don't want them to subconsciously limit the feature.

Reply:
Cool! That is all correct... I've tweaked the example.